File: Packs/CommonScripts/Scripts/CalculateTimeDifference/CalculateTimeDifference.py
import demistomock as demisto
from CommonServerPython import *

from CommonServerUserPython import *

start_time = demisto.args()["start_time"].replace('"', "")
end_time = demisto.args()["end_time"].replace('"', "")

try:
    # Strip microseconds and convert to datetime object
    start_time_obj = datetime.strptime(start_time.split(".")[0], "%Y-%m-%dT%H:%M:%S")
    end_time_obj = datetime.strptime(end_time.split(".")[0], "%Y-%m-%dT%H:%M:%S")
    # Calculate the difference in minutes
    time_diff = end_time_obj - start_time_obj
    mins = round((time_diff.total_seconds() / 60), 2)

    hr = f"Calculated Time Difference: {mins!s} minutes."
    context = {"Time.Difference": mins, "Time.Start": start_time, "Time.End": end_time}

    return_outputs(hr, context, mins)


except Exception as ex:
    return_error("Error occurred while parsing output from command. Exception info:\n" + str(ex))

File: Packs/CommonScripts/Scripts/CalculateTimeDifference/CalculateTimeDifference.js
 try {
      var start_time = args.start_time.replace(/"/g, "");
      var end_time = args.end_time.replace(/"/g, "");

      // Strip microseconds and parse the datetime
      var startTimeObj = new Date(start_time.split(".")[0]);
      var endTimeObj = new Date(end_time.split(".")[0]);

      // Calculate the difference in minutes
      var timeDiff = (endTimeObj - startTimeObj) / 1000; // difference in seconds
      var mins = Math.round((timeDiff / 60) * 100) / 100; // round to 2 decimal places


      ec = {
          'Time': {
              Difference: Number(mins),
              Start: start_time,
              End: end_time
          }
      }

      return {
          Type: entryTypes.note,
          Contents: Number(mins),
          ContentsFormat: formats.json,
          HumanReadable: `Calculated Time Difference: ${mins} minutes.`,
          ReadableContentsFormat: formats.markdown,
          EntryContext: ec
      };


  } catch (ex) {
      return {
          ContentsFormat: formats.markdown,
          Type: entryTypes.error,
          Contents: "Error occurred while parsing output from command. Exception info:\n" + ex.toString()
      };

  }